source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -62,16 +62,20 @@ void MutationUtils::applyHeaderMutations(const HeaderMutation& mutation, Http::H
       // something sensitive like the Authorization header.
       ENVOY_LOG(debug, "Ignorning improper attempt to set header {}", sh.header().key());
     } else {
-      // Make "false" the default. This is logical and matches the ext_authz
-      // filter. However, the router handles this same protobuf and uses "true"
-      // as the default instead.
-      const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
+      envoy::config::core::v3::HeaderValueOption::HeaderAppendAction append_action =
```

Comment:
Don't you need to support the new behavior here also? Can you move this logic into shared code so it's shared in both places with proper error handling, etc.?

Reply:
@mattklein123 I moved the code to get the append action by evaluating both the `append` as well as `append_action` flags in a utility class. Is that what you meant by "support the new behavior here"?